src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerTerminationValidator.java

```diff
@@ -54,16 +56,37 @@ public ApiErrors validate(HttpServletRequest request, OfficerFilingDto dto, Stri
 
         // Retrieve data objects required for the validation process
         final Transaction transaction = transactionService.getTransaction(transId, passthroughHeader);
-        final AppointmentFullRecordAPI companyAppointment = companyAppointmentService.getCompanyAppointment(transaction.getCompanyNumber(), dto.getReferenceAppointmentId(), passthroughHeader);
+        final Optional<AppointmentFullRecordAPI> companyAppointment = getValidatedOfficerAppointment(request, errorList, dto, transaction, passthroughHeader);
```

Comment:
Hi Phil, we are not calling the companyAppointmentService.getCompanyAppointment here as if the company appointment is not found, we want to fail early. This would mean returning the API Error list and not continuing with any of the other validation rules. We could do this validation check in this method instead of creating another method, like I did, but as it is a validation check required to return a specific message (DACT-300) the consistency is kept in that regard. We thought it was best that the error handling for this message is not completed in the method that is designed only to call each validation check. The other calls do not currently have any sort of validation check that was specified in a ticket. Does this make sense?

Reply:
Ok, I'm not sure I do following the reasoning.
Though I query the 'fail early' idea, why not return other possible errors with the termination date as well?

Reply:
From my understanding, other validation put in place will fail due to the lack of a company appointment. For example, verifying that the reference_etag from the dto, and the etag from the company appointment match. Null pointer exceptions will be caused later down the line if we continue. 
The getOfficerAppointment() is there to catch the exception companyAppointmentService.getCompanyAppointment would throw and handle the required error message outside of the validate method.
We could check the presence of this object in the calls that require it alternatively?

Reply:
nope, ok I hadn't twigged the inclusion of `AppointmentFullRecordAPI` in `validateTerminationDateAfterIncorporationDate`.  Though the error message in there seems strange.  Argh.. I'll ignore that.